### PR TITLE
feat(gax): add a common HTTP response error

### PR DESCRIPTION
**Pull request by codyoss**
Updates: #175

### CODE REVIEW

File: gax/src/error/error.rs [hunk 1]
```diff
@@ -63,8 +63,19 @@ impl Error {
...
     /// inner value if it is of type `T`, or `None` if no such inner value is
     /// found.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use gax::error::Error;
+    /// # use gax::error::HttpError;
+    /// # use std::collections::HashMap;
+    /// let error: Error = HttpError::new(404, HashMap::new(), None).into();
```
Comment (coryan) [inline gax/src/error/error.rs:73]: Is there a symbolic name for `404`?  though I wonder what is clearer... no action required.

File: gax/src/error/error.rs [hunk 2]
```diff
@@ -87,6 +98,12 @@ impl std::error::Error for Error {
     }
 }
 
+impl From<super::http_error::HttpError> for Error {
```
Comment (coryan) [inline gax/src/error/error.rs:101]: No change required, but should we prefer `super` or `crate`?

File: gax/src/error/rpc/mod.rs [hunk 3]
```diff
@@ -250,6 +251,14 @@ impl<'de> Deserialize<'de> for Code {
     }
 }
 
+impl TryFrom<HttpError> for Status {
```
Comment (coryan) [inline gax/src/error/rpc/mod.rs:254]: nice.

File: gax/tests/errors.rs [hunk 4]
```diff
@@ -66,3 +69,49 @@ fn downcast() -> Result<(), Box<dyn std::error::Error>> {
...
+            {"violations": [{"type": "type", "subject": "subject", "description": "desc"}]},
+        ]
+    });
+    let http_err = HttpError::new(
+        400,
+        HashMap::from_iter([("content-type".to_string(), "application/json".to_string())]),
+        Some(json.to_string().into()),
+    );
+
+    let status: Status = http_err.try_into()?;
```
Comment (coryan) [inline gax/tests/errors.rs:111]: Should we have a test for `try_into()` failures?  More specifically: do we want to verify it fails to convert something like:

```html
<!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 404 (Not Found)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
  </style>
  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  <p><b>404.</b> <ins>That’s an error.</ins>
  <p>The requested URL <code>/v7/projects/coryan-test/instances/test001/tables/test:readRows</code> was not found on this server.  <ins>That’s all we know.</ins>
  ```

**Review by coryan — APPROVED** [review]
A few questions, none of them blocking I think.

File: gax/tests/errors.rs (hunk 4 above)
Comment (codyoss) [reply gax/tests/errors.rs:111]: added a failure case.

File: gax/src/error/error.rs (hunk 1 above)
Comment (codyoss) [reply gax/src/error/error.rs:73]: Kept as 404 so we don't need an import. I think this is clear enough for now.

File: gax/src/error/error.rs (hunk 2 above)
Comment (codyoss) [reply gax/src/error/error.rs]: changed to crate

**Pull request merged**